components/calendar/demo/notice-calendar.ts

```diff
@@ -3,84 +3,79 @@ import { Component } from '@angular/core';
-    }
-
-    .notes-month {
-      text-align: center;
-      font-size: 28px;
-    }
+<nz-calendar>
+  <ul *nzDateCell="let date" class="events">
+     <ng-container *ngFor="let event of objects; trackBy:getEvents">
+        <li nz-popover *ngIf="event.date.getTime() === date.getTime()"
+         nzTitle="{{event.title}}" nzContent="{{event.content}}">
+           <nz-badge [nzStatus]="event.type" [nzText]="event.title"></nz-badge>
+        </li>
+     </ng-container>
+  </ul>
+  <ul *nzMonthCell="let month" class="events">
+     <ng-container *ngFor="let event of getMonthEvents(month)">
+       <li nz-popover nzTitle="{{event.title}}" nzContent="{{event.content}}">
+           <nz-badge [nzStatus]="event.type" [nzText]="event.title"></nz-badge>
+       </li>
+     </ng-container>
+  </ul>
+</nz-calendar>
+`,
+  styles: [`
+  .events {
+    text-align: center;
+    list-style: none;
+    margin: 0;
+    padding: 0;
+  }
 
-    .notes-month section {
-      font-size: 28px;
-    }
-  ` ]
+  .events .ant-badge-status {
+    overflow: hidden;
+    white-space: nowrap;
+    width: 100%;
+    text-overflow: ellipsis;
+    font-size: 12px;
+  }
+` ]
 })
 export class NzDemoCalendarNoticeCalendarComponent {
-  listDataMap = {
-    eight : [
-      { type: 'warning', content: 'This is warning event.' },
-      { type: 'success', content: 'This is usual event.' }
-    ],
-    ten   : [
-      { type: 'warning', content: 'This is warning event.' },
-      { type: 'success', content: 'This is usual event.' },
-      { type: 'error', content: 'This is error event.' }
-    ],
-    eleven: [
-      { type: 'warning', content: 'This is warning event' },
-      { type: 'success', content: 'This is very long usual event........' },
-      { type: 'error', content: 'This is error event 1.' },
-      { type: 'error', content: 'This is error event 2.' },
-      { type: 'error', content: 'This is error event 3.' },
-      { type: 'error', content: 'This is error event 4.' }
-    ]
-  };
+  current = new Date();
+  tomorrow = new Date();
+  yesterday = new Date();
+  nextMonth = new Date();
+
+  constructor() {
+    this.tomorrow.setDate(this.tomorrow.getDate() + 1);
+    this.yesterday.setDate(this.yesterday.getDate() - 1);
+    this.nextMonth.setMonth(this.nextMonth.getMonth() + 1);
+  }
+
+  objects =
+  [{
+    type: 'warning', content: 'This is warning event', title: 'This is title to warning event',
+    date: this.tomorrow
+  },
+  {
+    type: 'success', content: 'This is very long usual event with very very long decription...',
+    title: 'This is title to event with long description', date: this.current
+  },
+  {
+    type: 'error', content: 'This is error event', title: 'This is title to error event',
+    date: this.yesterday
+  },
+  {
+    type: 'default', content: 'This is default event', title: 'This is title to default event',
+    date: this.nextMonth
+  }];
+ 
+  getEvents(index, event) {
+    event.date.setHours(0, 0, 0, 0);
```

Comment:
should not set data in trackByFn

Reply:
When I did this example, I did not notice it, thank you, it's corrected.